src/main/java/cashleh/parser/Parser.java

```diff
@@ -82,38 +85,55 @@ public Command parse(String input) throws CashLehParsingException {
     }
 
     private Expense getExpense(String input) throws CashLehParsingException {
-        String[] format = {ADD_EXPENSE, "/amt", "/date:optional"};
+        String[] format = {ADD_EXPENSE, "/amt", "/date:optional", "/cat:optional"};
         HashMap<String, String> inputDetails = StringTokenizer.tokenize(input, format);
         String expenseName = inputDetails.get(ADD_EXPENSE);
         String expenseAmtString = inputDetails.get("/amt");
         String expenseDateString = inputDetails.get("/date");
+        String expenseCategoryString = inputDetails.get("/cat");
 
         if (expenseName.isEmpty()) {
             throw new CashLehParsingException(
                 "Oopsie! An expense without a description is like a CashLeh transaction without its story - not as fun!"
             );
         }
+
         double expenseAmt;
         try {
             expenseAmt = Double.parseDouble(expenseAmtString);
         } catch (NumberFormatException e) {
             throw new CashLehParsingException("Please enter a valid expense amount!");
         }
 
-        // default to current date if no date is specified
-        if (expenseDateString == null || expenseDateString.isEmpty()) {
+        LocalDate parsedDate = null;
+        if (!(expenseDateString == null || expenseDateString.isEmpty())) {
+            parsedDate = DateParser.parse(expenseDateString);
+        }
```

Comment:
I think the boolean expression within can be simplified using de morgan's, the current implementation is a bit hard to understand logically